contrib/win32/win32compat/win32_dirent.c

```diff
@@ -62,15 +121,64 @@ DIR * opendir(const char *name)
    Return 0 if successful, -1 if not.  */
 int closedir(DIR *dirp)
 {
-    if ( dirp && (dirp->hFile) ) {
+    if (!dirp)
+        return 0;
+    if (dirp->hFile)
         _findclose( dirp->hFile );
```

Comment:
You say it seriously?
After call `free(dirp);` read of field **hFile** cause an exception!